pkg/microservice/aslan/core/log/service/log.go

```diff
@@ -108,10 +116,64 @@ func getContainerLogFromS3(pipelineName, filenamePrefix string, taskID int64, lo
 		log.Errorf("GetContainerLogFromS3 Download err:%v", err)
 		return "", err
 	}
+
 	containerLog, err := ioutil.ReadFile(tempFile)
 	if err != nil {
 		log.Errorf("GetContainerLogFromS3 Read file err:%v", err)
 		return "", err
 	}
 	return string(containerLog), nil
 }
+
+func GetPodLogByHttp(podName, containerName, envName, productName string, tail int64, log *zap.SugaredLogger) (string, error) {
+	productInfo, err := commonrepo.NewProductColl().Find(&commonrepo.ProductFindOptions{Name: productName, EnvName: envName})
+	if err != nil {
+		log.Errorf("kubeCli.GetContainerLogStream error: %v", err)
+		return "", err
+	}
+	clientSet, err := kube.GetClientset(productInfo.ClusterID)
+	if err != nil {
+		log.Errorf("failed to find ns and kubeClient: %v", err)
+		return "", err
+	}
+
+	logOptions := &corev1.PodLogOptions{
+		Container: containerName,
+		TailLines: &tail,
+	}
+	ctx := context.TODO()
+	req := clientSet.CoreV1().Pods(productInfo.Namespace).GetLogs(podName, logOptions)
+	podLogs, err := req.Stream(ctx)
+	if err != nil {
+		log.Errorf("req.Stream error: %v,productInfo.Namespace:%v,podName:%v,containerName:%v", err,
+			productInfo.Namespace, podName, containerName)
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		log.Errorf("io.Copy error: %v", err)
+		return "", err
+	}
+	str := buf.String()
```

Comment:
the logic is almost same with function GetContainerLogs in pkg/tool/kube/containerlog/log.go, try to reuse it